### PR TITLE
Improve the support for NTP servers

### PR DESCRIPTION
**Pull request by poncovka**
Use the class TimeSourceData to represent a time source.

Modify the Timezone module to work with the DBus structures for time sources
instead of the strings. Rename the DBus property NTPServers to TimeSources.

Clean up the code for the NTP configuration.

The Timezone module should handle the timesource kickstart command.

**Depends on:** https://github.com/pykickstart/pykickstart/pull/322

### CODE REVIEW

File: pyanaconda/ntp.py [hunk 1]
```diff
@@ -41,27 +45,68 @@
...
+        server.hostname,
+        states.get_status_description(server)
+    )
 
-    :param server: hostname or IP address of an NTP server
-    :type server: string
-    :return: True if the given server is reachable and working, False otherwise
-    :rtype: bool
 
+def get_ntp_servers_summary(servers, states):
```
Comment (VladimirSlavik) [inline pyanaconda/ntp.py:78]: Unless I'm mistaken, this is not used anywhere?

**Comment by poncovka** [comment]
Jenkins, test this please.

File: pyanaconda/ntp.py (hunk 1 above)
Comment (poncovka) [reply pyanaconda/ntp.py:78]: It is used in TUI.

**Comment by jkonecny12** [comment]
jenkins, test this please

**Comment by poncovka** [comment]
Jenkins, test this please.

File: pyanaconda/ntp.py [hunk 2]
```diff
@@ -75,118 +73,89 @@ def ntp_server_working(server):
...
+        try:
             new_conf_file = open(out_file_path, "w")
-        else:
+        except IOError as ioerr:
+            msg = "Cannot open new config file {} for writing ({})."
+            raise NTPconfigError(msg.format(out_file_path, ioerr.strerror))
+    else:
+        try:
             (fildes, temp_path) = tempfile.mkstemp()
             new_conf_file = os.fdopen(fildes, "w")
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/ntp.py]: Fixed.

File: pyanaconda/startup_utils.py [hunk 3]
```diff
@@ -373,3 +372,24 @@ def parse_kickstart(ks, addon_paths, strict_mode=False):
...
     return ksdata
+
+
+def start_chronyd():
+    """Start the NTP daemon chronyd.
+
+    Set up NTP servers and start NTP daemon if not requested otherwise.
+    """
+    if not conf.system.can_set_time_synchronization:
+        return
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/startup_utils.py:384]: Fixed. Do you mean a decorator specifically for the time synchronization? I think that we should eventually move all this code to the Timezone module. Then the service activation could be handled in one place.

File: pyanaconda/ui/gui/spokes/datetime_spoke.py [hunk 4]
```diff
@@ -1161,16 +1134,24 @@ def on_ntp_switched(self, switch, *args):
             self.clear_info()
 
     def on_ntp_config_clicked(self, *args):
-        self._config_dialog.refresh()
+        servers = copy.deepcopy(self._ntp_servers)
```
[the remark this answers is not in the document]
Comment (poncovka) [reply pyanaconda/ui/gui/spokes/datetime_spoke.py:1137]: If you click the Cancel button in the dialog for NTP servers, the installer should forget the changes done in the dialog.

**Comment by poncovka** [comment]
Updated.

File: pyanaconda/startup_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (jkonecny12) [reply pyanaconda/startup_utils.py:384]: I was thinking about something more general for more or less all the confs? Not sure really.

File: pyanaconda/ui/gui/spokes/datetime_spoke.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (jkonecny12) [reply pyanaconda/ui/gui/spokes/datetime_spoke.py:1137]: I see, it's for the dialog. OK, thanks for explanation.

**Pull request merged**